src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,213 @@
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String insertRequest =
+                "INSERT INTO cars (model, manufacturer_id) VALUES (?, ?);";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createCarStatement = connection.prepareStatement(insertRequest,
+                        Statement.RETURN_GENERATED_KEYS)) {
+
+            createCarStatement.setString(1, car.getModel());
+            createCarStatement.setLong(2, car.getManufacturer().getId());
+            createCarStatement.executeUpdate();
+            ResultSet generatedKey = createCarStatement.getGeneratedKeys();
+            if (generatedKey.next()) {
+                Long id = generatedKey.getObject(1, Long.class);
+                car.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't insert from to DB", e);
+        }
+        insertDrivers(car);
+        return car;
+    }
+
+    @Override
+    public Car get(Long id) {
+        String selectRequest =
+                "SELECT c.id as car_id, model, m.id as manufacturer_id, m.name, m.country "
+                        + "FROM CARS c JOIN manufacturers m "
+                        + "ON c.manufacturer_id = m.id "
+                        + "WHERE c.id = ? AND c.is_deleted = FALSE;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarStatement =
+                        connection.prepareStatement(selectRequest)) {
+            getCarStatement.setLong(1, id);
+            ResultSet resultSet = getCarStatement.executeQuery();
+            if (resultSet.next()) {
+                car = parseCarWithManufacturerResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't find car in DB by id " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversFromCar(id));
+        }
+        return car;
+    }
+
+    @Override
+    public List<Car> getAll() {
+        Car car;
+        List<Car> carList = new ArrayList<>();
+        String selectGetAllRequest = "SELECT c.id as car_id, model, m.id as manufacturer_id,"
+                + " m.name, m.country FROM cars c JOIN manufacturers m "
+                + "ON c.manufacturer_id = m.id AND c.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsStatement = connection.prepareStatement(
+                        selectGetAllRequest, Statement.RETURN_GENERATED_KEYS)) {
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                car = parseCarWithManufacturerResultSet(resultSet);
+                car.setDrivers(getDriversFromCar(car.getId()));
```

Comment:
do not open a new connection inside the connection. extract getting drivers from try-catch